### PR TITLE
Exposing registering observer factory to make CFP more Spring friendly

### PR DESCRIPTION
**Pull request by abinav2307**
Exposing registerObserverFactory as a replacement for registerObserver. This now facilitates using objects in the ChangeFeedObserverFactory as Singleton objects.

### CODE REVIEW

File: src/test/com/microsoft/azure/documentdb/changefeedprocessor/Sample.java [hunk 1]
```diff
@@ -71,8 +71,9 @@ public static void main (String[]args) {
 
         ChangeFeedEventHost host = new ChangeFeedEventHost("hostname", docInfo, docAux, options, hostOptions );
 
+        IChangeFeedObserverFactory factory = new ChangeFeedObserverFactory(TestChangeFeedObserver.class);
         try {
-            host.registerObserver(TestChangeFeedObserver.class);
+            host.registerObserverFactory(factory);
```
Comment (milismsft) [inline src/test/com/microsoft/azure/documentdb/changefeedprocessor/Sample.java]: Can we keep the original code in here?

File: src/test/com/microsoft/azure/documentdb/changefeedprocessor/SimpleTest.java [hunk 2]
```diff
@@ -70,8 +70,10 @@ public void testCreateChangeFeedHostUsingSecrets()  {
         ChangeFeedEventHost host = new ChangeFeedEventHost("hotsname", docInfo, docAux, options, new ChangeFeedHostOptions() );
         Assert.assertNotNull(host);
 
+        IChangeFeedObserverFactory factory = new ChangeFeedObserverFactory(TestChangeFeedObserver.class);
+        
         try {
-            host.registerObserver(TestChangeFeedObserver.class);
+            host.registerObserverFactory(factory);
```
Comment (milismsft) [inline src/test/com/microsoft/azure/documentdb/changefeedprocessor/SimpleTest.java]: Same here.

File: src/test/com/microsoft/azure/documentdb/changefeedprocessor/Sample.java (hunk 1 above)
Comment (abinav2307) [reply src/test/com/microsoft/azure/documentdb/changefeedprocessor/Sample.java]: Done.

File: src/test/com/microsoft/azure/documentdb/changefeedprocessor/SimpleTest.java (hunk 2 above)
Comment (abinav2307) [reply src/test/com/microsoft/azure/documentdb/changefeedprocessor/SimpleTest.java]: Done.

**Pull request merged**